tests/test_project/tests/utils.py

```diff
@@ -0,0 +1,160 @@
+from django.contrib.auth import get_user_model
+from django.contrib.staticfiles.testing import StaticLiveServerTestCase
+
+User = get_user_model()
+
+
+class SeleniumTestCase(StaticLiveServerTestCase):
+    def open(self, url, driver=None):
+        """
+        Opens a URL
+        Argument:
+            url: URL to open
+            driver: selenium driver (default: cls.base_driver)
+        """
+        if not driver:
+            driver = self.web_driver
+        driver.get(f'{self.live_server_url}{url}')
+
+    def _create_user(self, **kwargs):
+        opts = dict(
+            username='tester',
+            password='tester',
+            first_name='Tester',
+            last_name='Tester',
+            email='test@tester.com',
+        )
+        opts.update(kwargs)
+        user = User(**opts)
+        user.full_clean()
+        return User.objects.create_user(**opts)
+
+    def _create_admin(self, **kwargs):
+        opts = dict(
+            username='admin', email='admin@admin.com', is_superuser=True, is_staff=True
+        )
+        opts.update(kwargs)
+        return self._create_user(**opts)
+
+    def login(self, username=None, password=None, driver=None):
+        """
+        Log in to the admin dashboard
+        Argument:
+            driver: selenium driver (default: cls.web_driver)
+            username: username to be used for login (default: cls.admin.username)
+            password: password to be used for login (default: cls.admin.password)
+        """
+        if not driver:
+            driver = self.web_driver
+        if not username:
+            username = self.admin_username
+        if not password:
+            password = self.admin_password
+        driver.get(f'{self.live_server_url}/admin/login/')
+        if 'admin/login' in driver.current_url:
+            driver.find_element_by_name('username').send_keys(username)
+            driver.find_element_by_name('password').send_keys(password)
+            driver.find_element_by_xpath('//input[@type="submit"]').click()
```

Comment:
preferably we will have a single source of all strings, i.e if we want to change `/admin/login` it should be a one place change.
I failed to do it in docker-openwisp when I made tests for that, but perhaps we can start having that repository now, so that changes in future are easier, what do you think?

Reply:
I agree with you. We can use it in other modules